### PR TITLE
Fix: block not working in site editor due to iframed editor

### LINKED ISSUE #93: Test against WordPress 5.9
**Is your enhancement related to a problem? Please describe.**
 Once WordPress 5.9 is released, we'll want to test Block for Apple Maps to see if any incompatibility issues arise.

**Describe the solution you'd like**
- [ ] test Block for Apple Maps on WordPress 5.9
- [ ] open issues for any incompatibilities noted in testing
- [ ] resolve issues identified in testing
- [ ] bump "tested up to" version
- [ ] if code changes needed due to incompatibilities, ship a plugin release, otherwise use "Plugin asset/readme update" action to update "tested up to" version on .org repo

**Designs**
n/a

**Describe alternatives you've considered**
none

**Additional context**
Related: #78, #82.

### LINKED ISSUE #100: Update block to `apiVersion: 2`
By upgrading block to `apiVersion: 2`, we can use `useBlockProps` and `userBlockProps.save()` for edit and save component. We also can add global style using Block Supports API, but we may not need it for this block.

### LINKED ISSUE #101: Tracking Issue: Site Editor Support
- [x] #102
- [x] #103

### LINKED ISSUE #102: Block not rendering in the iframed Site Editor
The Map block shows an error when inserted on a page inside the iframed site editor.

### LINKED ISSUE #103: Block not responding to mouse events like dragging the map canvas when rendered in the iframed Site Editor
This was discovered as a result of fixing #102 in #99. The Map itself does not respond to user interactions like dragging when it is rendered inside an iframe.

I think to verify this we should start by rendering a map inside an iframe outside of the editor to see whether it has something to do with mapkit or with WordPress.

### PR DESCRIPTION
**Pull request by fabiankaegy**
### Description of the Change

The Maps block did not work in the Site Editor which was introduced in WordPress 5.8. This came down to the way the mapkit js package checks whether the node that is passed to the `mapkit.Map` initialsation is a valid node. It does that check by verifying that the node is an instance of `window.Node`. This creates an issue though because the node is not created by the `window` object in which the JS runs. It was created by the `window` of the iframe. 

Screen Shot 2021-10-21 at 11 53 58

This PR also contains updates to the packages that I needed to run in order to get the local setup to work properly. So the actually relevant changes happen in 5f2a94b and 09f70c6.

### Checklist:

- [ ] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/components/AppleMap.js [hunk 1]
```diff
@@ -74,21 +67,36 @@ class AppleMap {
...
+		fakeTarget.classList.add('mapkit-wrapper');
+
+		const mapTargetElement = isInIframe ? fakeTarget : this.element;
+
+		this.map = new Map(mapTargetElement, this.mapOptions);
+		this.map._impl.zoomLevel = Number(zoom) || 15;
+
+		if (isInIframe) {
+			this.element.appendChild(mapTargetElement);
+		}
```
Comment (fabiankaegy) [inline src/components/AppleMap.js]: @dinhtungdu I would love to get your thoughts on this very hacky solution and see whether we can come up with a better implementation that does the same :)

Comment (dinhtungdu) [reply src/components/AppleMap.js]: Talked about it on the call but commenting here for reference. I can't think of any better solution than what you have here. For updating the map, I'm even okay with reinitializing the map after users update it.

The best solution would be asking the Mapkit team to update the library, but it won't be quick.

**Comment by fabiankaegy** [comment]
After some more digging I think it is all fixed now and can be tested in all editor contexts. This also closes #100 since it made working with the ref a bunch easier.

Tagging @dinhtungdu @jeffpaul @barneyjeffries for testing :)

**Review by dinhtungdu — COMMENTED** [review]
@fabiankaegy Great work on this PR 🎉 . I can edit the setting, add/remove/edit markers. But I'm having an issue with the alignment control. Every time I change the alignment, the map turns blank and it won't show up until the next reload. Please check my screencast for more detail.

https://user-images.githubusercontent.com/5423135/149889567-1269c17d-5024-49b9-ab46-e45a9e145f33.mov

<details>
<summary>Site Health Info</summary>

```
### wp-core ###

version: 5.9-RC2-52594
site_language: en_US
user_language: en_US
timezone: +00:00
permalink: undefined
https_status: true
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: open
environment_type: local
user_count: 2
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /Users/tenup/Sites/oss/app/public
wordpress_size: 49.90 MB (52325190 bytes)
uploads_path: /Users/tenup/Sites/oss/app/public/wp-content/uploads
uploads_size: 122.22 MB (128152057 bytes)
themes_path: /Users/tenup/Sites/oss/app/public/wp-content/themes
themes_size: 24.50 MB (25688534 bytes)
plugins_path: /Users/tenup/Sites/oss/app/public/wp-content/plugins
plugins_size: 2.57 GB (2761524561 bytes)
database_size: 3.00 MB (3145728 bytes)
total_size: 2.77 GB (2970836070 bytes)

### wp-active-theme ###

name: Twenty Twenty-Two (twentytwentytwo)
version: 1.0
author: the WordPress team
author_website: https://wordpress.org/
parent_theme: none
theme_features: core-block-patterns, post-thumbnails, responsive-embeds, editor-styles, html5, automatic-feed-links, block-templates, widgets-block-editor, wp-block-styles, editor-style
theme_path: /Users/tenup/Sites/oss/app/public/wp-content/themes/twentytwentytwo
auto_update: Disabled

### wp-themes-inactive (13) ###

Emptytheme: version: 1.0, author: the WordPress team, Auto-updates disabled
TT1 Blocks: version: 0.4.8, author: the WordPress team, Auto-updates disabled
Twenty Eleven: version: 3.9, author: the WordPress team, Auto-updates disabled
Twenty Fifteen: version: 3.0, author: the WordPress team, Auto-updates disabled
Twenty Fourteen: version: 3.2, author: the WordPress team, Auto-updates disabled
Twenty Nineteen: version: 2.1, author: the WordPress team, Auto-updates disabled
Twenty Seventeen: version: 2.8, author: the WordPress team, Auto-updates disabled
Twenty Sixteen: version: 2.5, author: the WordPress team, Auto-updates disabled
Twenty Ten: version: 3.5, author: the WordPress team, Auto-updates disabled
Twenty Thirteen: version: 3.4, author: the WordPress team, Auto-updates disabled
Twenty Twelve: version: 3.5, author: the WordPress team, Auto-updates disabled
Twenty Twenty: version: 1.8, author: the WordPress team, Auto-updates disabled
Twenty Twenty-One: version: 1.4, author: the WordPress team, Auto-updates disabled

### wp-plugins-active (2) ###

Block for Apple Maps: version: 1.0.1, author: 10up, Auto-updates disabled
User Switching: version: 1.5.8, author: John Blackbourn & contributors, Auto-updates disabled

### wp-plugins-inactive (17) ###

Accessibility CheckUp: version: 1.1.9, author: 10up, Auto-updates disabled
Akismet Anti-Spam: version: 4.2.1, author: Automattic, Auto-updates disabled
Autoshare for Twitter: version: 1.0.5, author: 10up (latest version: 1.0.6), Auto-updates disabled
Classic Editor: version: 1.6.2, author: WordPress Contributors, Auto-updates disabled
ClassifAI: version: 1.7.0, author: 10up, Auto-updates disabled
Distributor: version: 1.6.6-dev, author: 10up Inc., Auto-updates disabled
ElasticPress: version: 3.6.2, author: 10up (latest version: 3.6.6), Auto-updates disabled
Fox Admin Step Up MFA: version: 1.0.0-dev, author: 10up, Auto-updates disabled
Gutenberg: version: 11.7.1, author: Gutenberg Team (latest version: 12.3.2), Auto-updates disabled
miniOrange 2 Factor Authentication: version: 5.4.44, author: miniOrange (latest version: 5.4.50), Auto-updates disabled
Publisher Media Kit: version: 0.9.0, author: 10up (latest version: 1.0.0), Auto-updates disabled
Restricted Site Access: version: 7.2.0, author: Jake Goldman, 10up, Oomph, Auto-updates disabled
Retro Winamp Block: version: 0.1.0, author: 10up (latest version: 1.0.1), Auto-updates disabled
Simple Local Avatars: version: 2.2.0, author: Jake Goldman, 10up, Auto-updates disabled
Simple Podcasting: version: 1.2.0, author: 10up (latest version: 1.2.1), Auto-updates disabled
Sophi: version: 1.0.8-dev, author: 10up (latest version: 1.0.8), Auto-updates disabled
WP 2FA - Two-factor authentication for WordPress: version: 1.7.1, author: WP White Security (latest version: 2.1.0), Auto-updates disabled

### wp-media ###

image_editor: WP_Image_Editor_GD
imagick_module_version: Not available
imagemagick_version: Not available
imagick_version: Not available
file_uploads: File uploads is turned off
post_max_size: 1000M
upload_max_filesize: 300M
max_effective_size: 300 MB
max_file_uploads: 20
gd_version: bundled (2.1.0 compatible)
gd_formats: GIF, JPEG, PNG, BMP
ghostscript_version: not available

### wp-server ###

server_architecture: Darwin 21.1.0 x86_64
httpd_software: nginx/1.16.0
php_version: 8.0.0 64bit
php_sapi: fpm-fcgi
max_input_variables: 4000
time_limit: 1200
memory_limit: 256M
max_input_time: 600
upload_max_filesize: 300M
php_post_max_size: 1000M
curl_version: 7.77.0 (SecureTransport) LibreSSL/2.8.3
suhosin: false
imagick_availability: false
pretty_permalinks: true
htaccess_extra_rules: false

### wp-database ###

extension: mysqli
server_version: 8.0.16
client_version: mysqlnd 8.0.0
max_allowed_packet: 16777216
max_connections: 151

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /Users/tenup/Sites/oss/app/public/wp-content
WP_PLUGIN_DIR: /Users/tenup/Sites/oss/app/public/wp-content/plugins
WP_MEMORY_LIMIT: 40M
WP_MAX_MEMORY_LIMIT: 256M
WP_DEBUG: true
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: true
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: local
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable
mu-plugins: writable

```

</details>

**Comment by fabiankaegy** [comment]
@dinhtungdu Thanks for finding that. There was an issue when the DOM node was destroyed and a new one was created that the map instance was not updated this is now fixed 👍

**Review by dinhtungdu — COMMENTED** [review]
@fabiankaegy With the latest commit, the map block only shows the loading indicator in the Site Editor. The map is loaded normally in the Blocks Editor and front-end. 

image

**Comment by fabiankaegy** [comment]
@dinhtungdu Interesting 🤔 I am not able to replicate that currently running the WordPress 5.9 RC3. Will try with earlier WP versions.

**Comment by jeffpaul** [comment]
@dinhtungdu what setup are you using for that test, curious if maybe the php/app/db/browser versions might be causing a different result than what @fabiankaegy is using?

**Comment by fabiankaegy** [comment]
@jeffpaul We jumped on a call this morning and identified some issues together :) I have since pushed some more fixes and am feeling pretty good about it at the moment. So hoping for another round of thorough testing since I needed to change more code than I originally had hoped to... 

And also I now have quite a few follow up enhancements that I would like to make to the block like the search dropdown styling and the search component in general.

**Review by dinhtungdu — COMMENTED** [review]
@fabiankaegy  With the latest update, the block works flawlessly after being added to the Site Editor, all settings and alignment options are working as expected 🎉 . But I got some issues editing templates/template-parts containing the Apple Map Block:

Given that we added the map block to the footer template part and saved everything.

- Open the Site Editor from Appearance > Editor. The index template which contains the footer template part will be loaded. The map block only shows the spinner. To show the map, I need to select the block to trigger the map initialization event.
- Assuming that we don't have the loading issue above, the map is loaded correctly (in the Footer area) after opening the Site Editor. If I change to edit the Footer template part, the map block will be broken (This block has encountered an error and cannot be previewed.)


https://user-images.githubusercontent.com/5423135/150391960-5b9f54e7-a491-48d1-aa3d-c17a8a5e3120.mov

<details>
<summary>Site Health Info</summary>

```
### wp-core ###

version: 5.9-RC3
site_language: en_US
user_language: en_US
timezone: +00:00
permalink: undefined
https_status: true
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: open
environment_type: local
user_count: 2
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /Users/tenup/Sites/oss/app/public
wordpress_size: 49.90 MB (52325190 bytes)
uploads_path: /Users/tenup/Sites/oss/app/public/wp-content/uploads
uploads_size: 122.22 MB (128152057 bytes)
themes_path: /Users/tenup/Sites/oss/app/public/wp-content/themes
themes_size: 24.50 MB (25688534 bytes)
plugins_path: /Users/tenup/Sites/oss/app/public/wp-content/plugins
plugins_size: 2.57 GB (2761524561 bytes)
database_size: 11.00 MB (11534336 bytes)
total_size: 2.77 GB (2979224678 bytes)

### wp-active-theme ###

name: Twenty Twenty-Two (twentytwentytwo)
version: 1.0
author: the WordPress team
author_website: https://wordpress.org/
parent_theme: none
theme_features: core-block-patterns, post-thumbnails, responsive-embeds, editor-styles, html5, automatic-feed-links, block-templates, widgets-block-editor, wp-block-styles, editor-style
theme_path: /Users/tenup/Sites/oss/app/public/wp-content/themes/twentytwentytwo
auto_update: Disabled

### wp-themes-inactive (13) ###

Emptytheme: version: 1.0, author: the WordPress team, Auto-updates disabled
TT1 Blocks: version: 0.4.8, author: the WordPress team, Auto-updates disabled
Twenty Eleven: version: 3.9, author: the WordPress team, Auto-updates disabled
Twenty Fifteen: version: 3.0, author: the WordPress team, Auto-updates disabled
Twenty Fourteen: version: 3.2, author: the WordPress team, Auto-updates disabled
Twenty Nineteen: version: 2.1, author: the WordPress team, Auto-updates disabled
Twenty Seventeen: version: 2.8, author: the WordPress team, Auto-updates disabled
Twenty Sixteen: version: 2.5, author: the WordPress team, Auto-updates disabled
Twenty Ten: version: 3.5, author: the WordPress team, Auto-updates disabled
Twenty Thirteen: version: 3.4, author: the WordPress team, Auto-updates disabled
Twenty Twelve: version: 3.5, author: the WordPress team, Auto-updates disabled
Twenty Twenty: version: 1.8, author: the WordPress team, Auto-updates disabled
Twenty Twenty-One: version: 1.4, author: the WordPress team, Auto-updates disabled

### wp-plugins-active (2) ###

Block for Apple Maps: version: 1.0.1, author: 10up, Auto-updates disabled
User Switching: version: 1.5.8, author: John Blackbourn & contributors, Auto-updates disabled

### wp-plugins-inactive (17) ###

Accessibility CheckUp: version: 1.1.9, author: 10up, Auto-updates disabled
Akismet Anti-Spam: version: 4.2.1, author: Automattic, Auto-updates disabled
Autoshare for Twitter: version: 1.0.5, author: 10up (latest version: 1.0.6), Auto-updates disabled
Classic Editor: version: 1.6.2, author: WordPress Contributors, Auto-updates disabled
ClassifAI: version: 1.6.0, author: 10up, Auto-updates disabled
Distributor: version: 1.6.6-dev, author: 10up Inc., Auto-updates disabled
ElasticPress: version: 3.6.2, author: 10up (latest version: 3.6.6), Auto-updates disabled
Fox Admin Step Up MFA: version: 1.0.0-dev, author: 10up, Auto-updates disabled
Gutenberg: version: 11.7.1, author: Gutenberg Team (latest version: 12.4.0), Auto-updates disabled
miniOrange 2 Factor Authentication: version: 5.4.44, author: miniOrange (latest version: 5.4.51), Auto-updates disabled
Publisher Media Kit: version: 0.9.0, author: 10up (latest version: 1.0.0), Auto-updates disabled
Restricted Site Access: version: 7.2.0, author: Jake Goldman, 10up, Oomph, Auto-updates disabled
Retro Winamp Block: version: 0.1.0, author: 10up (latest version: 1.0.1), Auto-updates disabled
Simple Local Avatars: version: 2.2.0, author: Jake Goldman, 10up, Auto-updates disabled
Simple Podcasting: version: 1.2.0, author: 10up (latest version: 1.2.1), Auto-updates disabled
Sophi: version: 1.0.8-dev, author: 10up (latest version: 1.0.8), Auto-updates disabled
WP 2FA - Two-factor authentication for WordPress: version: 1.7.1, author: WP White Security (latest version: 2.1.0), Auto-updates disabled

### wp-media ###

image_editor: WP_Image_Editor_GD
imagick_module_version: Not available
imagemagick_version: Not available
imagick_version: Not available
file_uploads: File uploads is turned off
post_max_size: 1000M
upload_max_filesize: 300M
max_effective_size: 300 MB
max_file_uploads: 20
gd_version: bundled (2.1.0 compatible)
gd_formats: GIF, JPEG, PNG, BMP
ghostscript_version: not available

### wp-server ###

server_architecture: Darwin 21.1.0 x86_64
httpd_software: nginx/1.16.0
php_version: 8.0.0 64bit
php_sapi: fpm-fcgi
max_input_variables: 4000
time_limit: 1200
memory_limit: 256M
max_input_time: 600
upload_max_filesize: 300M
php_post_max_size: 1000M
curl_version: 7.77.0 (SecureTransport) LibreSSL/2.8.3
suhosin: false
imagick_availability: false
pretty_permalinks: true
htaccess_extra_rules: false

### wp-database ###

extension: mysqli
server_version: 8.0.16
client_version: mysqlnd 8.0.0
max_allowed_packet: 16777216
max_connections: 151

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: /Users/tenup/Sites/oss/app/public/wp-content
WP_PLUGIN_DIR: /Users/tenup/Sites/oss/app/public/wp-content/plugins
WP_MEMORY_LIMIT: 40M
WP_MAX_MEMORY_LIMIT: 256M
WP_DEBUG: true
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: true
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: local
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable
mu-plugins: writable

```

</details>

**Comment by fabiankaegy** [comment]
@dinhtungdu Thank you for testing this again :) 

I had forgotten to link the ticket here... I actually created a ticket on the Gutenberg repo with a bug that is causing the first issue you are describing: https://github.com/WordPress/gutenberg/issues/38110 

We are using the `useRefEffect` function to get access to the actual window of the iframe the block is being rendered in. And in the release post for the iframe editor it mentioned that the block gets rerendered once the scripts in the iframe are loaded so that you can access them inside the effect. 

This sadly does not appear to be happening in the template part editor currently which is why the spinner only disappears once you interact with the block itself. 

As for the second issue I will have to investigate that 👀

**Review by dinhtungdu — APPROVED** [review]
Great great work @fabiankaegy 🎉 !

This is working great on my end! I agree that we should accept known issues related to script loading and template switching. We can come back to address those issues later.

I notice that we're changing the code style. Look like the standard we're using isn't the WordPress recommended one. I think we should follow WP standard instead:

```js
# .prettierrc.js
module.exports = require( '@wordpress/prettier-config' );
```

Pre-approving here so you can just merge it after fixing the coding style : ).

**Comment by fabiankaegy** [comment]
@dinhtungdu Yeah I am waiting for the next release of the `@wordpress/scripts` package for our linting rules to apply again and then all this should get reformatted again... :O 

Right now it is still broken and therefore the coding style is off...

**Comment by dinhtungdu** [comment]
@fabiankaegy Then this PR is ready. :shipit:

**Pull request merged**